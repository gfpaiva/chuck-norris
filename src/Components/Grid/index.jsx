import React from 'react';
import PropTypes from 'prop-types';

import './Grid.scss';
import Card from '../Card';

export default function Grid({ items, children }) {
  return (
    <div className="grid">
      {(items && items.length) ? (
        items.map((item, idx) => (
          // eslint-disable-next-line react/no-array-index-key
          <Card key={idx}>
            {item}
          </Card>
        ))
      ) : (
        children
      )}
    </div>
  );
}

Grid.defaultProps = {
  items: [],
  children: null,
};

Grid.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};
