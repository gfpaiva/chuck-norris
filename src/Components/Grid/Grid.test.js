import React from 'react';
import { mount } from 'enzyme';

import Grid from './index';

describe('<Grid />', () => {
  it('should mount with children properly', () => {

    const wrapper = mount(
      <Grid>
          <p>
            Test
          </p>
      </Grid>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should mount with item prop properly', () => {

    const wrapper = mount(<Grid items={['Test']} />);

    expect(wrapper).toMatchSnapshot();
  });
});
