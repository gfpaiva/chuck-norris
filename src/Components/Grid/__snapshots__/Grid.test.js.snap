// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should mount with children properly 1`] = `
<Grid
  items={Array []}
>
  <div
    className="grid"
  >
    <p>
      Test
    </p>
  </div>
</Grid>
`;

exports[`<Grid /> should mount with item prop properly 1`] = `
<Grid
  items={
    Array [
      "Test",
    ]
  }
>
  <div
    className="grid"
  >
    <Card
      className=""
      hover={false}
      href=""
      key="0"
    >
      <div
        className="card"
      >
        Test
      </div>
    </Card>
  </div>
</Grid>
`;
